### PR TITLE
project internal changes

### PR DESCRIPTION
**Pull request by Abhishekism9450**
Support of project Internal API.

### CODE REVIEW

**Comment by abhimutant** [comment]
/ok-to-test pc
>Acceptance test run status: failure
>

**Comment by abhimutant** [comment]
/ok-to-test pc
>Acceptance test run status: failure
>Current line coverage (44.3) is below threshold (50). Kindly add more acceptance tests.

**Comment by abhimutant** [comment]
/ok-to-test pc
>Acceptance test run status: failure
>Current line coverage (6.3) is below threshold (50). Kindly add more acceptance tests.

**Comment by abhimutant** [comment]
/ok-to-test pc
>Acceptance test run status: failure
>Current line coverage (0.0) is below threshold (50). Kindly add more acceptance tests.

File: nutanix/resource_nutanix_project.go [hunk 1]
```diff
@@ -264,49 +362,535 @@ func resourceNutanixProject() *schema.Resource {
 				Optional: true,
 				Computed: true,
 			},
+			"use_project_internal": {
```
Comment (siddharth-kulshrestha) [inline nutanix/resource_nutanix_project.go:369]: Do we still need this flag value?
What if user passes the other fields and mark this field as false?

Comment (siddharth-kulshrestha) [reply nutanix/resource_nutanix_project.go:369]: I see that you have added RequiredWith attribute but only with user list not with acps and other attributes.

File: nutanix/resource_nutanix_project.go [hunk 2]
```diff
@@ -264,49 +362,535 @@ func resourceNutanixProject() *schema.Resource {
...
+		spec := &v3.ProjectInternalSpec{}
+		metadata := &v3.Metadata{}
+		projDetails := &v3.ProjectDetails{}
+		accessControlPolicy := make([]*v3.AccessControlPolicyList, 0)
+
+		var clusterUUID string
+		if clusterID, ok := d.GetOk("cluster_uuid"); ok {
+			clusterUUID = clusterID.(string)
+		}
+		response, err := conn.V3.GetProjectInternal(ctx, d.Id())
```
Comment (siddharth-kulshrestha) [inline nutanix/resource_nutanix_project.go]: Will this API get called even if there are no ACPs set by the user?

File: nutanix/resource_nutanix_project.go [hunk 3]
```diff
@@ -370,68 +954,203 @@ func resourceNutanixProjectRead(ctx context.Context, d *schema.ResourceData, met
...
-	resp, err := conn.V3.UpdateProject(d.Id(), project)
-	if err != nil {
-		if strings.Contains(fmt.Sprint(err), "ENTITY_NOT_FOUND") {
-			d.SetId("")
+		if response.Spec != nil {
+			spec = response.Spec
+
+			if response.Spec.ProjectDetail != nil || response.Spec.AccessControlPolicyList != nil {
+				projDetails = response.Spec.ProjectDetail
+				// accessControlPolicy = response.Spec.AccessControlPolicyList
```
Comment (siddharth-kulshrestha) [inline nutanix/resource_nutanix_project.go]: Do we need this commented line of code?

File: nutanix/resource_nutanix_project.go [hunk 4]
```diff
@@ -569,3 +1284,504 @@ func expandMetadata(d *schema.ResourceData, kind string) *v3.Metadata {
...
+			acps.ACP = acpSpec
+			acpList[k] = acps
+		}
+
+		return acpList
+	}
+	return nil
+}
+
+func UpdateExpandAcpRM(pr []interface{}, res *v3.ProjectInternalIntentResponse, d *schema.ResourceData, meta interface{}, projectUUID string, clusterUUID string) []*v3.AccessControlPolicyList {
```
Comment (siddharth-kulshrestha) [inline nutanix/resource_nutanix_project.go:1761]: What is this function doing ?
Can you add some code comments on these functions?

**Review by siddharth-kulshrestha — CHANGES_REQUESTED** [review]

File: nutanix/resource_nutanix_project.go (hunk 2 above)
Comment (abhimutant) [reply nutanix/resource_nutanix_project.go]: Yes. It will be called.

File: nutanix/resource_nutanix_project.go (hunk 3 above)
Comment (abhimutant) [reply nutanix/resource_nutanix_project.go]: Noted. Will remove this commented line.

File: nutanix/resource_nutanix_project.go (hunk 4 above)
Comment (abhimutant) [reply nutanix/resource_nutanix_project.go:1761]: This is called when we have to update the acps (adding, deleting) in UpdateContext when projects_internal flag is set.
Will add comments.

**Comment by abhimutant** [comment]
/ok-to-test pc
>Acceptance test run status: failure
>Current line coverage (44.1) is below threshold (50). Kindly add more acceptance tests.

File: nutanix/resource_nutanix_project.go (hunk 1 above)
Comment (abhimutant) [reply nutanix/resource_nutanix_project.go:369]: 1. yes, based on flag user is allowed to create ACPs. If marked as false with other values, project API will be called. 
2.Added RequiredWith attribute.

**Comment by abhimutant** [comment]
/ok-to-test pc
>Acceptance test run status: failure
>Current line coverage (44.3) is below threshold (50). Kindly add more acceptance tests.

File: README.md [hunk 5]
```diff
@@ -44,12 +44,16 @@ The Terraform Nutanix provider is designed to work with Nutanix Prism Central an
...
 
+> For the 1.7.1 release of the provider it will have N-2 compatibility with the Prism Central APIs. This release was tested against Prism Central versions pc2022.6, pc2022.4.0.1 and pc2022.1.0.2.
+
 ### note
 With v1.6.1 release of flow networking feature in provider, IAMv2 setups would be mandate. 
 Also, there is known issue for access_control_policies resource where update would be failing. We are continuously tracking the issue internally.
 
 with v1.7.0 release of user groups feature in provider, pc version should be minimum 2022.1 to support organisational and saml user group. 
 
+With v1.7.1 release of project internal  in provider is supported. Note to use this, set "use_project_internal" to true.
```
Comment (siddharth-kulshrestha) [inline README.md]: Can we also add something like "it enables mapping of ACPs with projects"?

**Review by siddharth-kulshrestha — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (abhimutant) [reply README.md]: Sure.

**Pull request merged**